tests/test_throttling.py

```diff
@@ -159,6 +159,26 @@ def test_request_throttling_multiple_throttles(self):
         assert response.status_code == 429
         assert int(response['retry-after']) == 58
 
+    def test_handle_negative_throttle_value(self):
+        self.set_throttle_timer(MockView_DoubleThrottling, 0)
+        request = self.factory.get('/')
+        for dummy in range(24):
+            response = MockView_DoubleThrottling.as_view()(request)
+        assert response.status_code == 429
+        assert int(response['retry-after']) == 60
+
+        previous_rate = User3SecRateThrottle.rate
+        User3SecRateThrottle.rate = '1/sec'
+
+        for dummy in range(24):
+            response = MockView_DoubleThrottling.as_view()(request)
+
+        assert response.status_code == 429
+        assert int(response['retry-after']) == 60
+
+        # reset
+        User3SecRateThrottle.rate = previous_rate
```

Comment:
If this test fails, the rate isn't restored to its original value, causing other tests to fail erroneously. The simple fix is to wrap the block in a try/finally or some kind of context manager to ensure the rate is restored. However I'm wondering if there should be an example/test rate class that demonstrates the changing rates (might be useful for related tests in the future?.

Reply:
I added try/finally in 957b39fa, but I'm not quite sure what you mean with

>  However I'm wondering if there should be an example/test rate class that demonstrates the changing rates (might be useful for related tests in the future?.

to be honest. You mean a rate class that changes the rate "automatically" or by calling a function or so? E.g so that we don't re-use other tests rate class?

Reply:
I meant potentially having a class that would automatically change its rate, demonstrating the behavior. But honestly, the try/catch is fine.